tests/rptest/services/redpanda.py

```diff
@@ -1952,23 +1980,73 @@ def cluster_ready_replicas(cluster_name: str):
                 "-o=jsonpath='{.status.replicas}'"
             ]).decode())
 
-        for pod_name in pod_names:
-            self.restart_pod(pod_name, pod_timeout)
+        # Check cluster readiness after restart of all pods
+        self.check_cluster_readiness(cluster_name, expected_replicas,
+                                     pod_timeout)
 
+    def check_cluster_readiness(self, cluster_name: str,
```

Comment:
The idea to move this from test module to redpanda service, so that it could be reused by other tests as well if needed. Reusability. I updated commit and tested. We can call this inside a test.